### ISSUE TITLE
docker-machine scp does not use the SSH port specified in ~/.docker/machine/machines/<machine>/config.json

### ISSUE THREAD

**Issue by cyli** [docker/machine#3492]
I have the following in my machine config:

```
{
    "ConfigVersion": 3,
    "Driver": {
        "IPAddress": ...,
        "MachineName": ...,
        "SSHUser": "root",
        "SSHPort": 2222,
        "SSHKeyPath": "/Users/cyli/.docker/machine/machines/<machine>/id_rsa",
        "StorePath": "/Users/cyli/.docker/machine",
        "SwarmMaster": false,
        "SwarmHost": "tcp://0.0.0.0:3376",
        "SwarmDiscovery": "",
        "AccessToken": ...,
        "DropletID": ...,
        "DropletName": "",
        "Image": "ubuntu-15-10-x64",
        "Region": "nyc3",
        "SSHKeyID": 2013668,
        "SSHKeyFingerprint": "",
        "Size": "4GB",
        "IPv6": false,
        "Backups": false,
        "PrivateNetworking": false,
        "UserDataFile": ""
    },
    ...
```

`docker-machine ssh` successfully connects to my Droplet.

When I run `docker-machine scp myfile.txt <machine-name>:myfile.txt` I get the following:

```
Docker Machine Version:  0.7.0, build a650a40
Found binary path at /usr/local/bin/docker-machine
Launching plugin server for driver digitalocean
Plugin server listening at address 127.0.0.1:56808
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(<machine-name>) Calling .GetSSHKeyPath
(<machine-name>) Calling .GetSSHKeyPath
(<machine-name>) Calling .GetIP
(<machine-name>) Calling .GetSSHUsername
{/usr/bin/scp [/usr/bin/scp -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -3 -i /Users/cyli/.docker/machine/machines/<machine-name>/id_rsa myfile.txt root@104.131.33.10:myfile.txt] []  <nil> <nil> <nil> [] <nil> <nil> <nil> <nil> false [] [] [] [] <nil>}
lost connection
exit status 1
```

The port value for scp can be specified with a `-P` option on mac os and the flavors of ubuntu I've used.  Not sure if this is generalizable to all platforms supported by docker-machine.

**Comment by gittycat** [comment]
Yes, I'm seeing the same error with 1.12-rc2

**Comment by panjiesw** [comment]
This is also still happening in v0.8.0

**Comment by twang2218** [comment]
This is definitely a bug. I have the same issue.

I changed the default SSHD port from `22/tcp` to other ports, and set up a SSHD honeypot on port `22`. Then when I use `docker-machine scp`, I can see the log information on the SSHD honeypot logs, which listen on port `22`. And all other command, such as `env`, `ssh` works as expected. So, it's a bug in `docker-machine scp`.

My `docker-machine` version is `v0.8.1`:

```bash
$ docker-machine -v
docker-machine version 0.8.1, build 41b3b25
```

**Comment by nxtruong** [comment]
Is this bug fix in the current Docker for Mac (version 1.12.2-rc1-beta27 (build: 12496)?
I still have the same issue, where I used a custom SSH port instead of 22. docker-machine's ssh worked, but scp failed with error:

> lost connection
exit status 1

**Comment by twang2218** [comment]
@nxtruong I don't think so, the version of `docker-machine` in current docker for mac is `version 0.8.2, build e18a919`, which is not include most of the master patches. 

@nathanleclaire Could you check the tag `v0.8.2` on commit e18a919?  There isn't much patches on e18a919 since `v0.8.1`. I think `v0.8.2` should be tagged on commit 8d10bb1.

**Comment by nathanleclaire** [comment]
It's on `master` and will be in 0.9.0.  Arguably could have ported it for 0.8.2 but we wanted to get the Go version fix out ASAP, sorry about the inconvenience in the interim folks.

**Comment by joedragons** [comment]
FYI, I'm testing docker-machine version 0.9.0-rc1, build ed849a7 and I am now getting the lost connection/exit status 1 using docker-machine scp.  As far as I know I don't have a custom port (I never have configured anything in my docker-machine's config.json, just used out of the box.  
I tried the inverse example from https://docs.docker.com/machine/reference/scp/ and get an "exit status 1".

My google search brought me here and seeing something changed in 0.9.0 thought you might want to know.  If I should file elsewhere or can provide more detail, let me know.  I am going to try rolling back and if it works in older version, advice my company not to upgrade for now.

**Comment by nathanleclaire** [comment]
thanks for the comment @joedragons .  I also see issue (e.g., exit status 1) with things that should work so I'll look into it

**Comment by nathanleclaire** [comment]
@joedragons https://github.com/docker/machine/pull/3873 should fix it.  Thanks for the report!

**Comment by joedragons** [comment]
@nathanleclaire Finally got around to verifying, works for me.  Thanks so much!

**Comment by MichaelTheBrute** [comment]
I ran into this problem yesterday and found a workaround (or solution?) today.  This is specific to OS X Sierra and latest Docker and VirtualBox as of this writing. 

Going through the docker Getting Started docs (https://docs.docker.com/get-started/part4/#create-a-cluster), the command: 

`docker-machine scp docker-compose.yml myvm1:~ `

Was immediately returning:

```
Lost connection
Exit Status 1
```

Finally, I tracked it down to my /etc/ssh/ssh_config when I attempted a classic "scp" command without the docker-machine ssh handholding.  The "scp" command returned immediately, complaining about the following incompatible lines: 

```
GSSAPIKeyExchange no
GSSAPITrustDNS no
```

I commented those out, which relieved the issue and "docker-machine scp" was able to work as intended.  I did not find this solution posted elsewhere.  

My system specifics are pasted below.  Hopefully this helps someone out there.  

OS
===
OS X Sierra
System Version:	macOS 10.12.3 (16D32)
Kernel Version:	Darwin 16.4.0

Docker
======
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 21:43:09 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 21:43:09 2017
 OS/Arch:      linux/amd64
 Experimental: true

VirtualBox
================
Version 5.1.22 r115126
